fury/actors/tensor.py

```diff
@@ -0,0 +1,214 @@
+    big_vectors_2 = np.repeat(evec2, n_verts, axis=0)
+    attribute_to_actor(box_actor, big_vectors_2, 'evec2')
+    big_vectors_3 = np.repeat(evec3, n_verts, axis=0)
+    attribute_to_actor(box_actor, big_vectors_3, 'evec3')
+
+    # Start of shader implementation
+
+    vs_dec = \
+        """
+        in vec3 center;
+        in float scale;
+        in vec3 evals;
+        in vec3 evec1;
+        in vec3 evec2;
+        in vec3 evec3;
+
+        out vec4 vertexMCVSOutput;
+        out vec3 centerMCVSOutput;
+        out float scaleVSOutput;
+        out vec3 evalsVSOutput;
+        out mat3 tensorMatrix;
+        """
+
+    # Variables assignment
+    v_assign = \
+        """
+        vertexMCVSOutput = vertexMC;
+        centerMCVSOutput = center;
+        scaleVSOutput = scale;
+        """
+
+    # Normalization
+    n_evals = "evalsVSOutput = evals/(max(evals.x, max(evals.y, evals.z)));"
+        
+    # Values constraint to avoid incorrect visualizations
+    evals = "evalsVSOutput = clamp(evalsVSOutput,0.05,1);"
+        
+    # Scaling matrix
+    sc_matrix = \
+        """
+        mat3 S = mat3(1/evalsVSOutput.x, 0.0, 0.0,
+                      0.0, 1/evalsVSOutput.y, 0.0,
+                      0.0, 0.0, 1/evalsVSOutput.z);
+        """
+
+    # Rotation matrix
+    rot_matrix = "mat3 R = mat3(evec1, evec2, evec3);"
+        
+    # Tensor matrix
+    t_matrix = "tensorMatrix = inverse(R) * S * R;"
+
+    vs_impl = compose_shader([v_assign, n_evals, evals, sc_matrix, rot_matrix,
+                              t_matrix])
+
+    # Adding shader implementation to actor
+    shader_to_actor(box_actor, 'vertex', decl_code=vs_dec, impl_code=vs_impl)
+
+    fs_vars_dec = \
+        """
+        in vec4 vertexMCVSOutput;
+        in vec3 centerMCVSOutput;
+        in float scaleVSOutput;
+        in vec3 evalsVSOutput;
+        in mat3 tensorMatrix;
+
+        uniform mat4 MCVCMatrix;
+        """
+
+    # Importing the sphere SDF
+    sd_sphere = import_fury_shader(os.path.join('sdf', 'sd_sphere.frag'))
+
+    # SDF definition
+    sdf_map = \
+        """
+        float map(in vec3 position)
+        {
+            /*
+            As the scaling is not a rigid body transformation, we multiply
+            by a factor to compensate for distortion and not overestimate
+            the distance.
+            */
+            float scFactor = min(evalsVSOutput.x, min(evalsVSOutput.y,
+                                     evalsVSOutput.z));
+                                     
+            /*
+            The approximation of distance is calculated by stretching the
+            space such that the ellipsoid becomes a sphere (multiplying by
+            the transformation matrix) and then computing the distance to
+            a sphere in that space (using the sphere SDF).
+            */
+            return sdSphere(tensorMatrix * (position - centerMCVSOutput),
+                scaleVSOutput*0.48) * scFactor;
```

Comment:
I set the sphere (that is transformed into an ellipsoid) with a radius of 0.5 (half of the box) for the ellipsoid to occupy the maximum space in the box. There is a small issue when drawing ellipsoids with one or two axes where the values that are too small, for example (1, .3, .00003), for some reason the ellipsoid goes a bit out of the box, as you can see in the image, so that's why set a value close to 0.5, to avoid an erroneous visualization.
![image](https://github.com/fury-gl/fury/assets/31288525/eb8c8680-05e2-42d1-b002-3ac3b4b6915b)